crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -239,18 +239,19 @@ pub trait EthCall: Call + LoadPendingBlock {
 
         let precompiles = get_precompiles(env.handler_cfg.spec_id);
         let mut inspector = AccessListInspector::new(initial, from, to, precompiles);
-        let (result, env) = self.inspect(&mut db, env, &mut inspector)?;
 
-        match result.result {
-            ExecutionResult::Halt { reason, .. } => Err(match reason {
-                HaltReason::NonceOverflow => RpcInvalidTransactionError::NonceMaxValue,
-                halt => RpcInvalidTransactionError::EvmHalt(halt),
-            }),
-            ExecutionResult::Revert { output, .. } => {
-                Err(RpcInvalidTransactionError::Revert(RevertError::new(output)))
+        let (result, env) = match self.inspect(&mut db, env, &mut inspector) {
+            Ok((res, env)) => (res, env),
+            Err(e) => {
+                return Err(e.into());
             }
-            ExecutionResult::Success { .. } => Ok(()),
-        }?;
+        };
+
+        let error = match result.result {
+            ExecutionResult::Halt { reason, .. } => Some(format!("{:?}", reason)),
+            ExecutionResult::Revert { output, .. } => Some(format!("{:?}", output)),
+            ExecutionResult::Success { .. } => None,
+        };
```

Comment:
we can keep the code on main and use RevertError::new(output) for the revert case

Reply:
I will use the code regarding `(result, env)` from the main branch as is. 
Also, I made some changes to the Revert part and left a commit for it.